### PR TITLE
Update Register and Login pages

### PR DESCRIPTION
**Pull request by vvyn**

### CODE REVIEW

File: pages/register.tsx [hunk 1]
```diff
@@ -129,9 +129,9 @@ export default function Register({ allowedRegistrations }: RegisterPageProps) {
   }
 
   // disable this for testing
-  if (!user) {
-    router.push('/');
-  }
+  //if (!user) {
+  //  router.push('/');
+  //}
```
Comment (DedsecKnight) [inline pages/register.tsx:134]: Uncomment this.

File: pages/register.tsx [hunk 2]
```diff
@@ -272,38 +275,67 @@ export default function Register({ allowedRegistrations }: RegisterPageProps) {
...
+                      <span className="text-[#313131] poppins-semibold text-sm ml-2">
+                        I have read and agree to the{' '}
+                        <a
+                          href="https://static.mlh.io/docs/mlh-code-of-conduct.pdf"
+                          className="text-[#40B7BA] hover:underline"
+                        >
+                          MLH Code of Conduct
+                        </a>
+                      </span>
+                    </div>
```
Comment (DedsecKnight) [inline pages/register.tsx:327]: This should go inside `hackportal.config.ts` as one of the questions.

**Review by DedsecKnight — CHANGES_REQUESTED** [review]

File: hackportal.config.ts [hunk 3]
```diff
@@ -527,6 +571,8 @@ interface CheckboxQuestion extends RegistrationQuestion {
   options: Array<{
     title: string;
     value: string;
+    link: string;
+    linkText: string;
```
Comment (DedsecKnight) [inline hackportal.config.ts:575]: Thoughts on these 2 fields optional? Meaning the interface will look like the following: 
```ts
options: Array<{
  title: string;
  value: string;
  link?: string;
  linkText?: string;
}>
```
That way, we won't have to add `link` and `linkText` to every other textbox questions.

File: components/registerComponents/RegistrationQuestion.tsx [hunk 4]
```diff
@@ -89,35 +89,48 @@ function Question(props) {
...
+              <Field
+                type="checkbox"
+                name={props.question.name}
+                value={option.value}
+                className="rounded border-[#313131] border-2 cursor-pointer"
+              />
               &nbsp;{option.title}
+              <a href={option.link} className="text-[#40B7BA] hover:underline">
+                {option.linkText}
+              </a>
```
Comment (DedsecKnight) [inline components/registerComponents/RegistrationQuestion.tsx:120]: Thoughts on making `link` and `linkText` optional fields and then only render the anchor if `option.link` and `option.linkText` is not `undefined`?

File: hackportal.config.ts (hunk 3 above)
Comment (vvyn) [reply hackportal.config.ts:575]: Sounds good! Thank you, I was unsure of how to do that.

File: pages/register.tsx
[the remark this answers is not in the document]
Comment (salmanjaher) [reply pages/register.tsx:1]: Is it possible to make all the question elements the same UI style? 

All like this - Where the question is on top and the box is empty (no question)
image

This page is bothering me the most - first 2 question are a different style than the last
image

[the remark this answers is not in the document]
Comment (salmanjaher) [reply pages/register.tsx:1]: If we make this UI change you can throw the Required in after the question in small font like I suggested with the "optional" tag. (this should fix the form from breaking ideally)

[the remark this answers is not in the document]
Comment (salmanjaher) [reply pages/register.tsx:1]: feat: make the little bubbles on the bottom of the form clickable to move to different pages!

[the remark this answers is not in the document]
Comment (salmanjaher) [reply pages/register.tsx:1]: Also im not sure if this is supposed to be the case but the submit button doesn't do anything right now. Can we add like a
"Application submitted. Expect to hear back soon!"
[Picture of our mascot]

**Review by salmanjaher — CHANGES_REQUESTED** [review]
Can this page be changed from "Hacker Registration" to "Hacker Application"? Registration sounds like it guarantees acceptance.

File: pages/register.tsx [hunk 5]
```diff
@@ -321,47 +344,43 @@ export default function Register({ allowedRegistrations }: RegisterPageProps) {
               {/* Hackathon Questions */}
               {registrationSection == 2 && (
                 <section className="bg-white lg:w-3/5 md:w-3/4 w-full min-h-[35rem] mx-auto rounded-2xl md:py-10 py-6 px-8 mb-8 text-[#4C4950]">
-                  <h2 className="sm:text-2xl text-xl font-semibold sm:mb-3 mb-1">
+                  <h2 className="sm:text-2xl text-xl poppins-bold sm:mb-3 mb-1 mt-2">
                     Hackathon Experience
                   </h2>
-                  <div className="flex flex-col">
+                  <div className="flex flex-col poppins-regular md:px-4">
                     {hackathonExperienceQuestions.map((obj, idx) => (
```
[the remark this answers is not in the document]
Comment (DedsecKnight) [reply pages/register.tsx:402]: > Also is other supposed to add a new box?

Can you clarify a bit more what you mean by this?

[the remark this answers is not in the document]
Comment (salmanjaher) [reply pages/register.tsx:402]: It looks like the issue is fixed :)

**Review by salmanjaher — APPROVED** [review]
lgtm, thank you for the quick changes!

**Review by DedsecKnight — APPROVED** [review]

**Pull request merged**